config/tools/config_generator.cpp

```diff
@@ -0,0 +1,935 @@
+static constexpr auto tmpl_watchtower_log_level = "tmpl_watchtower_log_level";
+
+// All acceptable log levels
+std::set<std::string> log_levels
+    = {"TRACE", "DEBUG", "INFO", "WARN", "ERROR", "FATAL"};
+
+namespace cbdc::generate_config {
+
+    config_generator::config_generator(std::string& _template_config_file,
+                                       const size_t _start_port,
+                                       std::string _build_dir)
+        : m_template_config_file(_template_config_file),
+          m_current_port(_start_port) {
+        // Get Project root dir and build dir
+        std::filesystem::path build_dir = std::filesystem::current_path();
+        if(_build_dir.at(_build_dir.size() - 1) == '/') {
+            _build_dir.erase(_build_dir.end() - 1);
+        }
+        // This config generator class assumes project root is "opencbdc-tx"
+        while(build_dir.has_parent_path()) {
+            if(build_dir.filename() == "opencbdc-tx") {
+                m_project_root_dir = build_dir;
+                std::string delimiter = "/";
+                std::string tmp_str = _build_dir;
+                size_t pos = 0;
+                std::string token;
+                while((pos = tmp_str.find("/")) != std::string::npos) {
+                    token = tmp_str.substr(0, pos);
+                    build_dir = build_dir.append(token);
+                    tmp_str.erase(0, pos + delimiter.length());
+                }
+                token = tmp_str.substr(0, pos);
+                build_dir = build_dir.append(token);
+                tmp_str.erase(0, pos + delimiter.length());
+                m_build_dir = build_dir;
+                std::cout << "Build directory determined to be "
+                          << m_build_dir.string() << std::endl;
+                std::cout << "Project Root directory determined to be "
+                          << m_project_root_dir.string() << std::endl;
+                break;
+            } else {
+                build_dir = build_dir.parent_path();
+            }
+        }
+        template_file_is_valid = true;
+        if(!std::filesystem::exists(m_template_config_file)) {
+            template_file_is_valid = false;
+            std::filesystem::path temp_config_tools_dir = m_project_root_dir;
+            temp_config_tools_dir.append("config").append("tools");
+            std::filesystem::path temp_build_config_tools_dir = m_build_dir;
+            temp_build_config_tools_dir.append("config").append("tools");
+            std::cout << "Warning: File provided, " << m_template_config_file
+                      << ", does not exist. Attempting to copy it from its "
+                         "original location, "
+                      << temp_config_tools_dir.string() << " to "
+                      << temp_build_config_tools_dir.string() << std::endl;
+            copy_templates_to_build_dir();
+            // Try to use newly copied tmplate files
```

Comment:
```suggestion
            // Try to use newly copied template files
```
Might make sense to run `aspell` or something else to catch any simple typos.

Reply:
Fixed this. The older version of IDE i use used to have spellchecker plugin but it no longer works with newer version. Will find alternative (or use aspell as you suggested).

DONE